### PR TITLE
Docs: Adds new Enterprise Quickstart using Docker Compose

### PR DESCRIPTION
**Pull request by ZPain8464**
This guide updates the current Enterprise Quickstart with a Docker Compose implementation. I deleted the previous guide, but we can append it to the new guide if we want to include both implementations in our docs.

### CODE REVIEW

**Comment by desimone** [comment]
\cc @wasaga

**Comment by wasaga** [comment]
please exclude `k8s` folder from cspell

**Comment by ZPain8464** [comment]
@wasaga this skips cspell when I run it now, but cspell will ignore `k8s` going forward.

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (cmo-pomerium) [inline content/docs/releases/enterprise/install/quickstart.mdx:96]:
```suggestion
`pomerium console` requires the following services before the container will run:
```

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (cmo-pomerium) [inline content/docs/releases/enterprise/install/quickstart.mdx:132]:
```suggestion
The `healthcheck` parameter ensures the database is ready to receive requests. Without a healthcheck, other containers that depend on the database service (like Pomerium Console) will attempt to connect to the database before it is in a ready state, **which will crash the container**. (See the [Docker healthcheck docs](https://docs.docker.com/engine/reference/builder/#healthcheck) for more information.)
```

I just think this should be bolded

**Review by cmo-pomerium — APPROVED** [review]
minor change, LGTM

**Review by wasaga — CHANGES_REQUESTED** [review]
as discussed: 

1. merge individual snippets into single final files to prevent errors while executing the guide; as snippets currently are lacking hierarchy in few places and would not be easily merged into docker compose file 
2. hardcode secrets that are not user specific [similar to this](https://github.com/pomerium/documentation/blob/4438c609d3522cdc3523ac4542c309983c4bcb6e/content/docs/capabilities/high-availability.mdx?plain=1#L161-L163) 
3. move TLS discussion and Recovery JWT section to GUI Configuration page

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (alexrudd2) [inline content/docs/releases/enterprise/install/quickstart.mdx:33]: Suggestion:
~~credentials to access the private Cloudsmith Docker registry.~~
credentials to access the private Cloudsmith.io Docker registry, and a license key

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (alexrudd2) [inline content/docs/releases/enterprise/install/quickstart.mdx:60]: suggestion: 
```
    volumes:
        - ./config.yaml:/pomerium/config.yaml:ro # Mount your config file read-only
        - .data:/data:rw # Mount persistent persistent data storage
```

IMO this should be in the [main Quick-Start](https://www.pomerium.com/docs/quickstart) also.

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (alexrudd2) [inline content/docs/releases/enterprise/install/quickstart.mdx:87]: Suggestion: use `latest` instead of `v0.20.0`

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (alexrudd2) [inline content/docs/releases/enterprise/install/quickstart.mdx:39]: suggestion: link to https://docs.docker.com/network/bridge/

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (alexrudd2) [inline content/docs/releases/enterprise/install/quickstart.mdx:192]: suggestion:
~~You must use the same signing key in Pomerium Core and Enterprise.~~
You must use the same signing key and shared secret in Pomerium Core and Enterprise.

**Review by alexrudd2 — COMMENTED** [review]
As a customer of Pomerium, I was able to follow this guide more or less successfully to standup the Console with Docker Swarm.  (Thanks to @desimone for pointing me here)

Docker Swarm uses much of the Docker compose file syntax, with the following differences:
(1) Networks become `type: overlay` instead of `type: bridge`. (docker [docs](https://docs.docker.com/network/overlay/))
(2) `depends_on` is unfortunately [unsupported](https://github.com/moby/moby/issues/31333).  The Console will simply enter a crashloop until the services are available.
(3) There is no need to expose ports between services/containers on an overlay network.  Services can access all ports of other services by default.
(4) A more robust system for data persistence is needed if there are multiple hosts.  Filesystems mounts... aren't great.  I'm using a combination of GlusterFS and node pinning.

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (wasaga) [reply content/docs/releases/enterprise/install/quickstart.mdx:260]: the format was correct; it's a `host:port`

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (alexrudd2) [inline content/docs/releases/enterprise/install/quickstart.mdx:225]: "Multiple administrators are defined as a comma-separated string."

File: content/examples/enterprise-quickstart/docker-compose.yaml
Comment (wasaga) [inline content/examples/enterprise-quickstart/docker-compose.yaml:29]: maybe we should externalize all customizable env variables rather then expect the user to spot the "REPLACE ME" https://docs.docker.com/compose/environment-variables/#using-the---env-file--option

File: content/examples/enterprise-quickstart/docker-compose.yaml
Comment (wasaga) [inline content/examples/enterprise-quickstart/docker-compose.yaml:48]:
```suggestion
      test: [ "CMD-SHELL", "pg_isready -d pomeriumDB -U postgres" ]
```

we can use `postgres` user for demos

File: content/examples/enterprise-quickstart/config.yaml
Comment (wasaga) [inline content/examples/enterprise-quickstart/config.yaml:12]:
```suggestion
databroker_storage_connection_string: postgresql://postgres:postgres@postgres/<database>?sslmode=disable
```

**Review by wasaga — CHANGES_REQUESTED** [review]

File: content/examples/enterprise-quickstart/docker-compose.yaml
Comment (ZPain8464) [reply content/examples/enterprise-quickstart/docker-compose.yaml:29]: I think we should deploy this as is. Externalizing customizable variables would require adding a `.env` file with placeholder values. Either way, the user will need to fill in those values themselves, so some configuration is required. Thoughts?

**Comment by wasaga** [comment]
@ssveta7ak please test this guide using `v0.21.0-rc2` tags for core + console

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (wasaga) [inline content/docs/releases/enterprise/install/quickstart.mdx:70]: `pomerium` is not a valid URL schema, should be either `http` or `https`

**Comment by wasaga** [comment]
@ZPain8464 could you please also fix the Install page ? 

1. Quickstart is no longer RPM/DEB oriented 
2. Please link to Kustomize instead of helm 

image

File: content/examples/enterprise-quickstart/docker-compose.yaml
Comment (ssveta7ak) [inline content/examples/enterprise-quickstart/docker-compose.yaml:55]: Maybe we should just use postgres as a database name
```suggestion
      test: [ "CMD-SHELL", "pg_isready -d postgres -U postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5
      start_period: 30s
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=postgres
```

File: content/examples/enterprise-quickstart/docker-compose.yaml
Comment (ssveta7ak) [inline content/examples/enterprise-quickstart/docker-compose.yaml:40]: This file is not described in the Quickstart 
```suggestion
```

**Comment by ssveta7ak** [comment]
https://deploy-preview-256--pomerium-docs.netlify.app/   is not up to date.

File: content/examples/enterprise-quickstart/docker-compose.yaml
Comment (ssveta7ak) [inline content/examples/enterprise-quickstart/docker-compose.yaml:33]: Maybe we should use postgres as a database name otherwise it requires extra steps for the user (create a database with <DB_name>). 
```suggestion
      - DATABASE_URL=postgresql://<user>:<password>@database/postgres?sslmode=disable
```

File: content/examples/enterprise-quickstart/config.yaml
Comment (ssveta7ak) [inline content/examples/enterprise-quickstart/config.yaml:5]: `idp_provider_url: ` - compulsory element

Comment (ZPain8464) [reply content/examples/enterprise-quickstart/config.yaml:5]: @ssveta7ak this depends on your IdP. GitHub, for instance, doesn't require that, nor does Google. So the IdP configuration will vary.

**Comment by ZPain8464** [comment]
@wasaga is this ok to merge? If so, could you please approve the PR when you have time this week?

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (wasaga) [inline content/docs/releases/enterprise/install/quickstart.mdx:81]: this would fail trying to mount `./pomerium/metrics` as it does not exist. this may be made a blind volume:

```suggestion
      - metrics-data:/data:rw
```

you just need to also declare that volume at the root. https://docs.docker.com/compose/compose-file/#volumes

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (wasaga) [inline content/docs/releases/enterprise/install/quickstart.mdx:110]: see comment about prometheus metrics volume below. 

```suggestion
volumes:
  pgdata:
  metrics-data:
```

**Review by wasaga — CHANGES_REQUESTED** [review]
I'm not sure the files in the `example/` folder are referenced? please also see https://pomerium-com.slack.com/archives/C0277SH5KL3/p1676341990003349

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (ZPain8464) [reply content/docs/releases/enterprise/install/quickstart.mdx]: I've corrected this here: https://github.com/pomerium/documentation/pull/256/commits/f6be2733149abc1d1e93dada7d53e76dc9ecd160

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (wasaga) [inline content/docs/releases/enterprise/install/quickstart.mdx:64]:
```suggestion
    image: docker.cloudsmith.io/pomerium/enterprise/pomerium-console:v0.21.0
```

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (wasaga) [inline content/docs/releases/enterprise/install/quickstart.mdx:47]:
```suggestion
    image: pomerium/pomerium:v0.21.0
```

**Comment by ZPain8464** [comment]
@ssveta7ak I assigned you for review/testing. Just went through the guide with Denis, it's ready to go pending your approval!

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (ssveta7ak) [inline content/docs/releases/enterprise/install/quickstart.mdx:110]: When I try to do `docker compose up` I have an error `services.pomerium_console Additional property args is not allowed
`

**Review by ssveta7ak — CHANGES_REQUESTED** [review]

File: content/docs/releases/enterprise/install/quickstart.mdx
Comment (ssveta7ak) [inline content/docs/releases/enterprise/install/quickstart.mdx:82]:
```suggestion
```title="console-config.yaml"
```

**Review by ssveta7ak — CHANGES_REQUESTED** [review]

**Review by ssveta7ak — CHANGES_REQUESTED** [review]
Console doesn't start.
```
quickstart_docker_enterprise-pomerium_console-1  | {"level":"info","version":"0.21.0-1676045406+53d6bf83+2023-02-09T22:09:46Z","envoy_version":"1.24.0+05e1343657086268edaa16a1e8a1429d9c8f6e50a44f492cf629fdc71da2ecdc","time":"2023-02-16T17:26:58Z","message":"cmd/pomerium-console"}
quickstart_docker_enterprise-pomerium_console-1  | {"level":"fatal","error":"Get \"https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json\": x509: certificate signed by unknown authority","time":"2023-02-16T17:26:58Z"}
quickstart_docker_enterprise-pomerium_console-1 exited with code 1
```

**Review by ssveta7ak — APPROVED** [review]
Fixed Console not starting due to certificates not being trusted by explicitly providing a signing key.

**Review by wasaga — APPROVED** [review]

**Pull request merged**
